kaffeine/catalog_manager.go

```diff
@@ -163,37 +192,40 @@ func (cm *CatalogManager) RemoveCatalog(uri string) (oldFc FunctionCatalog, err
 	}
 
 	oldFc = cm.Catalogs[uri]
-
 	delete(cm.Catalogs, uri)
 
 	return oldFc, nil
 }
 
-// Clobbers catalog
+// Updates the catalog with the given uri. This is accomplished by deleting the
 func (cm *CatalogManager) UpdateCatalog(uri string) (oldFc FunctionCatalog, err error) {
-	// fmt.Println("UPDATING", uri)
-	// fmt.Println("REMOVING", uri)
 	oldFc, err = cm.RemoveCatalog(uri)
 	if err != nil {
 		return
 	}
 
-	// fmt.Println("GETTING EXTERNAL", uri)
+	revertToOldFc := func() {
```

Comment:
not usually a huge fan of closures inside a function. 

I would do potentially two things
1. make this it's own function or
2. Potentially add it in a recover block, so if something panics, it recovers the backup, https://gobyexample.com/recover